src/Rector/AbstractRector.php

```diff
@@ -131,8 +132,9 @@ final public function enterNode(Node $node): int|Node|null
         }
 
         $filePath = $this->file->getFilePath();
-        if ($this->skipper->shouldSkipCurrentNode($this, $filePath, static::class, $node)) {
-            return null;
+        $skipperResult = $this->skipper->shouldSkipCurrentNode($this, $filePath, static::class, $node);
+        if ($skipperResult !== SkipperResult::noSkip) {
+            return $skipperResult === SkipperResult::skipTree ? NodeVisitor::DONT_TRAVERSE_CHILDREN : null;
```

Comment:
It will skip all rules after rule visited on the node, which invalid, it only allowed on rector rules, which AbstractRector decorate the `refactor()` return:

https://github.com/rectorphp/rector-src/blob/912d0b7ecc78aae4cbc2f3df7992fab9b5c63014/src/Rector/AbstractRector.php#L156-L173

use return null after decorated.